src/redux/epics/download-epic.js

```diff
@@ -64,15 +65,12 @@ const beginDownload = (action$, store) => {
           tx => addrToIdx[tx.address]
         );
 
-        const encryptedFileContents = orderedTransactions
+        const bytesArray = orderedTransactions
           .map(tx => tx.signatureMessageFragment)
+          .map(msg => FileProcessor.decryptFile(msg, handle))
+          .filter(x => !!x) // Remove nulls
```

Comment:
This is where the magic happens

Reply:
Looks good, were you able to do any testing of this?  

It's hard to test it now, I imagine, because there's not a bunch of chunks in the datamap that won't decrypt.  

In a future PR, we should make some kind of datamap generator (for testing) that deliberately adds some chunks to the datamap that wouldn't decrypt with the handle, so we can know our logic works.  Could even be a checkbox on the page with the playground URL that will insert some chunks that won't decrypt into the datamap, then "download" them.

Reply:
Haven’t been able to test unfortunately. I agree though that we should be adding more extensive testing especially around the datamap logic